components/scream/src/share/io/scream_scorpio_interface.F90

```diff
@@ -1018,23 +1086,38 @@ end subroutine get_new_pio_atm_file
   !  grid_write_darray_1d_real: Write a variable defined on this grid
   !
   !---------------------------------------------------------------------------
-  subroutine grid_write_darray_1d_real(filename, hbuf, varname)
-
+  subroutine grid_write_darray_1d_real(filename, varname, var_data_ptr)
     ! Dummy arguments
-    character(len=*),          intent(in)    :: filename       ! PIO filename
-    real(rtype),               intent(in)    :: hbuf(:)
-    character(len=*),          intent(in)    :: varname
+    character(len=*),   intent(in) :: filename       ! PIO filename
+    character(len=*),   intent(in) :: varname
+    type (c_ptr),       intent(in) :: var_data_ptr
 
     ! Local variables
-    type(pio_atm_file_t), pointer              :: pio_atm_file
-    type(hist_var_t), pointer                :: var
-    integer                                  :: ierr
-    logical                                  :: found
+    real(rtype), dimension(:), pointer :: var_data
+    type(pio_atm_file_t), pointer      :: pio_atm_file
+    type(hist_var_t), pointer          :: var
+    integer                            :: ierr,var_size,ndims,i
+    logical                            :: found
 
     call lookup_pio_atm_file(trim(filename),pio_atm_file,found)
     call get_var(pio_atm_file,varname,var)
+
+    ! We don't want the extent along the 'time' dimension
+    if (var%has_t_dim) then
+      ndims = SIZE(var%dimlen)-1
```

Comment:
The if and else are the same statements here.

Reply:
Ctrl+C - Ctrl+V

Reply:
or for us VI users "yy" - "p"